src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/TextFormatting/NumberSubstitution.cs

```diff
@@ -508,7 +508,11 @@ private CultureInfo CreateTraditionalCulture(CultureInfo numberCulture, int firs
                     int n = firstDigit + i - 0x10000;
                     twoChars[0] = (char)((n >> 10) | 0xD800);     // high surrogate
                     twoChars[1] = (char)((n & 0x03FF) | 0xDC00);  // low surrogate
+#if NET
                     digits[i] = new string(twoChars);
```

Comment:
You can replace both sides with [Span.ToString](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Span.cs,368).